src/main/java/de/tum/in/www1/artemis/domain/TextCluster.java

```diff
@@ -48,6 +42,7 @@
     @OneToMany(mappedBy = "cluster")
     @OrderBy("position_in_cluster")
     @JsonIgnoreProperties("cluster")
+    @LazyCollection(LazyCollectionOption.FALSE)
```

Comment:
By removing `Transactional` context from `getTextSubmissionWithoutAssessment` method I need to fetch all required elements inside `findByParticipation_ExerciseIdAndResultIsNullAndSubmittedIsTrue` method. However, when I added `blocks.cluster.blocks` I receive `MultipleBagFetchException - cannot simultaneously fetch multiple bags` exception. To overcome this issue I used the solution provided here https://stackoverflow.com/questions/4334970/hibernate-throws-multiplebagfetchexception-cannot-simultaneously-fetch-multipl 
If there is a better way to fix it, please, let me know.

Reply:
Following the StackOverflow post, it is recommended to fetch the data in two queries. This prevents the problem. The two datasets then need to be merged in Java. I went ahead and implemented this in a19c3f2.